benchmarks/benchmark_result.py

```diff
@@ -83,5 +86,5 @@ def to_json(self):
         return json.dumps(self.to_dict())
 
     @classmethod
-    def from_json(cls, json_str):
+    def from_json(cls, json_str: str) -> "BenchmarkResult":
```

Comment:
```suggestion
    def from_json(cls, json_str: str) -> Self:
```

Reply:
Looks like this isn't a feature in python 3.11?

Reply:
ya fwiw this broke the macos tests, I switch to `from __future__`..